python/tvm/relay/testing/tflite.py

```diff
@@ -0,0 +1,152 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Common utilities for creating TFLite models"""
+from distutils.version import LooseVersion
+import numpy as np
+import pytest
+import tvm
+
+pytest.importorskip("tflite")
+pytest.importorskip("tensorflow")
+import tflite.Model  # pylint: disable=wrong-import-position
+import tensorflow as tf  # pylint: disable=wrong-import-position
+
+
+class TFLiteModel:
+    """Creates TFLite Model and facilitates reference data generation"""
+
+    def __init__(self, dtype):
+        self.serial_model = None  # This is what TFLite convert() provides
+        self.dtype = dtype  # This is the dtype of graph inputs
+        self.shape_dict = {}
+        self.dtype_dict = {}
+
+    @tf.function
+    def conv2d_single_function(self, ifm_tensor, args):
```

Comment:
I feel like creation of this function should be part of the test itself so its clear what's being tested, rather than hidden away in common testing utilities

Reply:
Yeah, I was thinking about it too. But wanted to wait until I had seen more cases that might be potential users of it. The problem with exposing this function is that the end user could chain this with other functions and would expect create_model() to work, but it might not because of the tf.function pragmas.

Reply:
As discussed offline, we will keep single layer creation as part of the infra to avoid duplication of this bit across multiple tests. If necessary, to include complex cases, user can write their own @tf.function and still create a model using this infra.